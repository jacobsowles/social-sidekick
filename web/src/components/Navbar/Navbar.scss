@import '@styles/_variables.scss';

.navbar {
  margin-bottom: 20px;
  border: 0;
  background: #ffffff;
  border-radius: 0;
  box-shadow: 0 1px 2px 0 rgba(60, 64, 67, 0.3), 0 2px 6px 2px rgba(60, 64, 67, 0.15);

  .navbar-brand {
    color: #252525;
    padding-top: 20px;
    padding-bottom: 20px;

    &:hover {
      color: $brand-primary;
    }

    svg {
      margin-right: 10px;
    }
  }

  .navbar-collapse.collapsing,
  .navbar-collapse.in {
    .navbar-nav > li.active > a {
      box-shadow: none;
    }

    ul:first-child {
      margin-bottom: 0;
    }

    .navbar-right {
      margin-top: 0;
    }
  }

  .navbar-nav {
    > li {
      > a {
        color: $gray-darker;
        padding: 20px;

        &:hover {
          color: $brand-primary;
          text-decoration: none;
        }
      }

      &.active > a {
        background: none;
        box-shadow: inset 0 -2px 0 0 $brand-primary;
        font-weight: bold;

        &:focus,
        &:hover {
          background: none;
        }

        &:hover {
          color: $brand-primary;
        }
      }

      > a {
        &:hover {
          color: $brand-primary;
        }
      }
    }
  }

  .profile-thumbnail {
    img {
      width: 35px;
      height: 35px;
      border-radius: 35px;
    }
  }

  .dropdown {
    ul li a {
      padding-left: 10px;
      padding-right: 10px;
    }

    .dropdown-header {
      margin-left: 0;
      padding-left: 10px;
    }

    .dropdown-menu .divider {
      margin: 5px 0;
    }

    #user-dropdown {
      padding-top: 12.5px;
      padding-bottom: 12.5px;
      padding-right: 10px;
    }
  }
}

@media (max-width: 767px) {
  .navbar {
    .navbar-brand {
      padding-top: 15px;
      padding-bottom: 15px;
    }

    .navbar-nav {
      > li > a {
        padding: 10px 20px;
      }

      .dropdown {
        #user-dropdown {
          padding-top: 10px;
          padding-bottom: 10px;
        }
      }
    }
  }
}
