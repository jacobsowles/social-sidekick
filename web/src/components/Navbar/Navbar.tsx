import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import { MenuItem, Nav, Navbar as BootstrapNavbar, NavDropdown, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';

import AuthService from '@core/auth';
import './Navbar.scss';

export interface NavbarProps {
  className?: string;
  handleLogout: () => void;
  picture?: string;
  username?: string;
}

class Navbar extends PureComponent<NavbarProps> {
  public static contextTypes = {
    router: PropTypes.object
  };

  public static defaultProps = {
    className: undefined as string,
    picture: undefined as string,
    username: undefined as string
  };

  private authService = new AuthService();

  public logout = () => {
    this.authService.logout((location: string) => this.context.router.history.replace(location));
    this.props.handleLogout();
  };

  public render() {
    return (
      <BootstrapNavbar className={classNames('navbar', this.props.className)}>
        <BootstrapNavbar.Header>
          <BootstrapNavbar.Brand>
            <a href="/">
              <FontAwesomeIcon icon="cube" />
              Social Sidekick
            </a>
          </BootstrapNavbar.Brand>
          <BootstrapNavbar.Toggle />
        </BootstrapNavbar.Header>
        <BootstrapNavbar.Collapse>
          <Nav className="nav-left">
            <LinkContainer to="/contact">
              <NavItem>Contact</NavItem>
            </LinkContainer>
          </Nav>
          <Nav pullRight>
            {this.authService.isAuthenticated() ? (
              <>
                <LinkContainer to="/dashboard">
                  <NavItem>Dashboard</NavItem>
                </LinkContainer>
                <LinkContainer to="/connections">
                  <NavItem>Connections</NavItem>
                </LinkContainer>
                <NavDropdown
                  eventKey={1}
                  id="user-dropdown"
                  title={
                    <span className="profile-thumbnail">
                      <img
                        src={this.props.picture || 'thumbnail-placeholder.gif'}
                        alt="thumbnail"
                      />
                    </span>
                  }
                >
                  <MenuItem header>Logged in as {this.props.username}</MenuItem>
                  <MenuItem divider />
                  <MenuItem eventKey={1.1} onClick={this.logout}>
                    <FontAwesomeIcon icon={['fas', 'sign-out-alt']} /> Log Out
                  </MenuItem>
                </NavDropdown>
              </>
            ) : (
              <NavItem onClick={this.authService.login}>Log In</NavItem>
            )}
          </Nav>
        </BootstrapNavbar.Collapse>
      </BootstrapNavbar>
    );
  }
}

export default Navbar;
