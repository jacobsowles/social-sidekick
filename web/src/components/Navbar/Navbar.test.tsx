import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import { Nav as BootstrapNav, Navbar as BootstrapNavbar } from 'react-bootstrap';

import { init, shallow, toJson } from '@tests/component-test-base';
import Navbar, { NavbarProps } from './Navbar';

describe('Navbar', () => {
  let props: NavbarProps;
  init(() => <Navbar {...props} />);

  beforeEach(() => {
    props = {
      className: undefined,
      handleLogout: jest.fn()
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  it('should always render a BootstrapNavbar component', () => {
    shallow()
      .find(BootstrapNavbar)
      .should.have.lengthOf(1);
  });

  it('should always render a BootstrapNavbar.Header component', () => {
    shallow()
      .find(BootstrapNavbar.Header)
      .should.have.lengthOf(1);
  });

  it('should always render a BootstrapNavbar.Brand component', () => {
    shallow()
      .find(BootstrapNavbar.Brand)
      .should.have.lengthOf(1);
  });

  it('should always render a BootstrapNavbar.Toggle component', () => {
    shallow()
      .find(BootstrapNavbar.Toggle)
      .should.have.lengthOf(1);
  });

  it('should always render a BootstrapNavbar.Collapse component', () => {
    shallow()
      .find(BootstrapNavbar.Collapse)
      .should.have.lengthOf(1);
  });

  it('should always render a FontAwesomeIcon component', () => {
    shallow()
      .find(FontAwesomeIcon)
      .should.have.lengthOf(1);
  });

  it('should always render a BootstrapNav component with the `pullLeft` set to true', () => {
    shallow()
      .find(BootstrapNav)
      .findWhere(nav => nav.prop('className') === 'nav-left')
      .should.have.lengthOf(1);
  });

  it('should always render a BootstrapNav component with the `pullRight` set to true', () => {
    shallow()
      .find(BootstrapNav)
      .findWhere(nav => nav.props().pullRight)
      .should.have.lengthOf(1);
  });

  describe('when `className` is defined', () => {
    beforeEach(() => {
      props.className = 'another-class';
    });

    it('should include the specified class name', () => {
      shallow()
        .first()
        .prop('className')
        .should.contain('another-class');
    });

    it('should include the navbar class name', () => {
      shallow()
        .first()
        .prop('className')
        .should.contain('navbar');
    });
  });

  describe('when `className` is not defined', () => {
    it('should only include the default class names', () => {
      shallow()
        .first()
        .prop('className')
        .should.equal('navbar');
    });
  });
});
