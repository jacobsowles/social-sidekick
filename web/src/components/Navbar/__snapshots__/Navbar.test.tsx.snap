// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component without crashing 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  className="navbar"
>
  <NavbarHeader>
    <NavbarBrand>
      <a
        href="/"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="cube"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        Social Sidekick
      </a>
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      className="nav-left"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/contact"
      >
        <NavItem
          active={false}
          disabled={false}
        >
          Contact
        </NavItem>
      </LinkContainer>
    </Nav>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <NavItem
        active={false}
        disabled={false}
        onClick={[Function]}
      >
        Log In
      </NavItem>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
